### PR TITLE
Load translations onLanguageChanged

### PR DESCRIPTION
**Pull request by CharString**
- Adds I18NManager to cantain all language string loading and signalling re-renders.
 - Moves `onLanguageChanged` callback from `LanguageSelection` params to `ConfigContext`
   
 Closes open-formulieren/open-forms#2255

### CODE REVIEW

**Comment by CharString** [comment]
Flatpickr seems to automagically change. I don't know what triggers it, but on every language change it will add a new node to the DOM with a new flatpickr script with the new locale. It even keeps working if I remove this line: 
https://github.com/open-formulieren/open-forms-sdk/blob/5e41548abd4ddae4e236909ae04ae43275d4db0a/src/sdk.js#L94
I tried to find the magic, because it doesn't clean up the previous nodes, but I gave up.

Also tried to get
https://github.com/open-formulieren/open-forms-sdk/blob/5e41548abd4ddae4e236909ae04ae43275d4db0a/src/i18n.js#L14-L21
to use `import()` such that it will load strings dynamically and async to reduce main bundle size. But `try {} catch {}` doesn't catch the import errors that can occur.

File: src/components/LanguageSelection/LanguageSelection.stories.mdx [hunk 1]
```diff
@@ -33,7 +33,6 @@ export const ConfigDecorator = (Story, { args }) => (
         type: { summary: 'number' },
       },
     },
-    onLanguageChanged: { table: { disable: true } },
```
Comment (sergei-maertens) [inline src/components/LanguageSelection/LanguageSelection.stories.mdx:36]: this arg is not relevant for the `LanguageSelectionDisplay` component and you can't change it in the controls anyway, that's why I disabled it for the canvas.

File: src/i18n.js [hunk 2]
```diff
@@ -1,19 +1,99 @@
...
 
 const loadLocaleData = (locale) => {
-    switch (locale) {
-        case 'nl':
-            return messagesNL;
-        default:
-            return messagesEN;
-    }
+  switch (locale) {
+    case "nl":
```
Comment (sergei-maertens) [inline src/i18n.js:16]: please keep the code style consistent with existing code style in the file - we'll implement prettier for the SDK at some point, but javascript strings are generally single quote, while JSX props use double quotes.

These sort of formatting changes really clutter the diff and make reviewing harder.

**Review by sergei-maertens — CHANGES_REQUESTED** [review]
I'm not a fan of hardcoding the App in this container component, why not use `props.children` instead?

We now have a component called `I18NManager` doing much more than that - it's also rendering the app, loading the form details from the API which is really the responsibility of other components that need to be composed.

I'll see if I can come to a set up that is more loosely coupled

**Pull request closed without merging**